schema-engine/cli/src/main.rs

```diff
@@ -4,10 +4,19 @@ mod commands;
 mod json_rpc_stdio;
 mod logger;
 
+use std::{backtrace::Backtrace, sync::Arc, time::Duration};
+
 use schema_connector::{BoxFuture, ConnectorHost, ConnectorResult};
-use schema_core::rpc_api;
-use std::{backtrace::Backtrace, sync::Arc};
+use schema_core::RpcApi;
 use structopt::StructOpt;
+use tokio::{signal, sync::oneshot};
+use tokio_util::sync::CancellationToken;
+
+/// The timeout for graceful shutdown of asynchronous tasks like network connections on SIGTERM.
+const GRACEFUL_SHUTDOWN_TIMEOUT: Duration = Duration::from_secs(4);
+
+/// The shutdown deadline for blocking background tasks.
+const BLOCKING_TASKS_SHUTDOWN_TIMEOUT: Duration = Duration::from_millis(200);
```

Comment:
None of these options need to be configured, right?

Reply:
I don't know. Probably not. I can add some env vars if you think it might be useful.